src/type-generator.ts

```diff
@@ -496,8 +504,7 @@ export class TypeGenerator {
       return { type: 'any', toJson: x => x };
     }
 
-    const comps = def.$ref.substring(prefix.length).split('.');
-    const typeName = TypeGenerator.normalizeTypeName(comps[comps.length - 1]);
```

Comment:
This was the offending code. Instead, we consider the entire namespace and normalize it.